deltacat/compute/compactor_v2/utils/compaction_utils.py

```diff
@@ -0,0 +1,713 @@
+            delete_file_envelopes,
+            task_max_parallelism,
+        )
+    logger.info(f"Got {len(merge_results)} merge results.")
+
+    merge_results_retrieved_at: float = time.time()
+    merge_end: float = time.monotonic()
+
+    total_dd_record_count = sum([ddr.deduped_record_count for ddr in merge_results])
+    total_deleted_record_count = sum(
+        [ddr.deleted_record_count for ddr in merge_results]
+    )
+    logger.info(
+        f"Deduped {total_dd_record_count} records and deleted {total_deleted_record_count} records..."
+    )
+
+    mutable_compaction_audit.set_input_records(total_input_records_count.item())
+
+    telemetry_time_merge = mutable_compaction_audit.save_step_stats(
+        CompactionSessionAuditInfo.MERGE_STEP_NAME,
+        merge_results,
+        merge_results_retrieved_at,
+        merge_invoke_end - merge_start,
+        merge_end - merge_start,
+    )
+
+    mutable_compaction_audit.set_records_deduped(total_dd_record_count.item())
+    mutable_compaction_audit.set_records_deleted(total_deleted_record_count.item())
+    return (
+        merge_results,
+        total_hb_record_count,
+        telemetry_time_hb,
+        total_dd_record_count,
+        total_deleted_record_count,
+        telemetry_time_merge,
+    )
+
+
+def _run_local_merge(
+    params: CompactPartitionParams,
+    uniform_deltas,
+    compacted_partition,
+    round_completion_info,
+    delete_strategy,
+    delete_file_envelopes,
+    mutable_compaction_audit,
+    previous_compacted_delta_manifest,
+    total_input_records_count,
+) -> tuple[list[Any], Any]:
+    local_merge_input: MergeInput = generate_local_merge_input(
+        params,
+        uniform_deltas,
+        compacted_partition,
+        round_completion_info,
+        delete_strategy,
+        delete_file_envelopes,
+    )
+    estimated_da_bytes = (
+        mutable_compaction_audit.estimated_in_memory_size_bytes_during_discovery
+    )
+    estimated_num_records: int = sum(
+        [
+            entry.meta.record_count
+            for delta in uniform_deltas
+            for entry in delta.manifest.entries
+        ]
+    )
+    local_merge_options = local_merge_resource_options_provider(
+        estimated_da_size=estimated_da_bytes,
+        estimated_num_rows=estimated_num_records,
+        total_memory_buffer_percentage=params.total_memory_buffer_percentage,
+        round_completion_info=round_completion_info,
+        compacted_delta_manifest=previous_compacted_delta_manifest,
+        ray_custom_resources=params.ray_custom_resources,
+        primary_keys=params.primary_keys,
+        memory_logs_enabled=params.memory_logs_enabled,
+    )
+    local_merge_result = ray.get(
+        mg.merge.options(**local_merge_options).remote(local_merge_input)
+    )
+    total_input_records_count += local_merge_result.input_record_count
+    merge_results = [local_merge_result]
+    return merge_results, total_input_records_count
+
+
+def _hash_bucket(
```

Comment:
The order for methods is not followed. To reduce scroll up and down, keep methods used in methods in topological order.

Reply:
The above ordering is correct. Just make sure it's reverse.